### PR TITLE
feat: Job label validation

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: [ISD-116](https://discourse.charmhub.io/t/specification-isd116-github-self-hosted-runners-reactive-scheduling/13755)

### Overview

Add label validation by case-insensitive comparison of the labels supported by the charm with the labels provided in the JobDetails of the message queue.

Successful integration test run: https://github.com/canonical/github-runner-operator/actions/runs/11037880494

### Rationale

This is to requeue incompatible messages. This is not something we currently expect (as it means the router has made an error while forwarding), but should be handled to catch programming errors.

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The library version in `pyproject.toml` is incremented

[ISD-116]: https://warthogs.atlassian.net/browse/ISD-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/github_runner_manager/reactive/consumer.py [hunk 1]
```diff
@@ -100,12 +105,35 @@ def consume(
...
+    """Validate the labels of the job.
+
+    Args:
+        labels: The labels of the job.
+        supported_labels: The supported labels for the runner.
+
+    Returns:
+        True if the labels are valid, False otherwise.
+    """
+    return all(label in supported_labels for label in labels)
```
Comment (jdkandersson) [inline src/github_runner_manager/reactive/consumer.py]: Perhaps make labels and supported labels a set and do a subset? That should be a much faster operation that this

File: tests/unit/reactive/test_consumer.py
Comment (jdkandersson) [inline tests/unit/reactive/test_consumer.py:1]: I think there are too few test cases here, think about the boundary cases for the set comparison of labels like empty sets, non-overlapping sets, partially overlapping, fully overlapping, fully overlapping where one is larger etc. Though perhaps not needed if you use inbuilt set operations

File: src/github_runner_manager/github_client.py [hunk 2]
```diff
@@ -269,14 +269,12 @@ def _to_job_info(job: dict) -> JobInfo:
...
-        # +00:00, which is not supported by datetime.fromisoformat
+        # datetime strings should be in ISO 8601 format, but they can also use Z instead of +00:00,
+        # which is not supported by datetime.fromisoformat
         created_at = datetime.fromisoformat(job["created_at"].replace("Z", "+00:00"))
         started_at = datetime.fromisoformat(job["started_at"].replace("Z", "+00:00"))
-        # conclusion could be null per api schema, so we need to handle that
-        # though we would assume that it should always be present,
-        # as the job should be finished
+        # conclusion could be null per api schema, so we need to handle that,
+        # though we would assume that it should always be present, as the job should be finished.
```
Comment (cbartz) [inline src/github_runner_manager/github_client.py:277]: Just reformatting the comments as requested by @yhaliaw

**Review by Thanhphan1147 — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]
